contracts/access/access-control/impls.rs

```diff
@@ -77,40 +66,32 @@ pub trait AccessControl: AccessControlStorage {
                 .insert(address, true);
             // TODO: Emit event
         }
-        Ok(())
     }
 
-    fn _revoke_role(
-        &mut self,
-        role: RoleType,
-        address: AccountId,
-    ) -> Result<(), AccessControlError> {
+    fn revoke_role(&mut self, role: RoleType, address: AccountId) {
         let caller = Self::env().caller();
-        self._check_role(&self._get_role_admin(&role), &caller)?;
-
+        self._check_role(&self._get_role_admin(&role), &caller);
         self._do_revoke_role(role, address);
-        Ok(())
     }
 
-    fn _renounce_role(
-        &mut self,
-        role: RoleType,
-        address: AccountId,
-    ) -> Result<(), AccessControlError> {
-        let caller = Self::env().caller();
-        if caller != address {
-            return Err(AccessControlError::InvalidCaller);
-        }
-
+    fn renounce_role(&mut self, role: RoleType, address: AccountId) {
+        assert_eq!(Self::env().caller(), address, "{}", AccessControlError::InvalidCaller.as_ref());
         self._do_revoke_role(role, address);
-        Ok(())
     }
 
-    fn _check_role(&self, role: &RoleType, address: &AccountId) -> Result<(), AccessControlError> {
-        if !self._has_role(role, address) {
-            return Err(AccessControlError::MissingRole);
-        }
-        Ok(())
+    fn set_role_admin(&mut self, role: RoleType, admin_role: RoleType) {
+        self._roles_mut()
+            .entry(role)
+            .or_insert_with(RoleData::default)
+            .admin_role = admin_role;
+        // TODO: Emit event
```

Comment:
Can we force users to override events, as we did for other contracts?

Reply:
We will do that, but in another PR